daemon/container_unix.go

```diff
@@ -948,7 +948,8 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 
 	ep, err := container.getEndpointInNetwork(n)
 	if err == nil {
-		return fmt.Errorf("container already connected to network %s", idOrName)
```

Comment:
Would it be an option to check if a container with the same name, but a different IP-address is trying to join?